File: Backend/src/repositories/menu/dish.ts
import { DataTypes, Model } from 'sequelize'
import sequelize from '../index'

class Dish extends Model {
  public id!: number;
  public name!: string;
  public description?: string;
  public meal_type!: 'cafe_manha' | 'almoco' | 'cafe_tarde' | 'janta'
}

Dish.init(
  {
    id: {
      type: DataTypes.INTEGER,
      autoIncrement: true,
      primaryKey: true
    },
    name: {
      type: DataTypes.STRING,
      allowNull: false
    },
    description: {
      type: DataTypes.STRING,
      allowNull: true
    },
    meal_type: {
      type: DataTypes.ENUM('cafe_manha', 'almoco', 'cafe_tarde', 'janta'),
      allowNull: false,
    },
  },
  {
    sequelize,
    modelName: 'Dish',
    tableName: 'dish',
    timestamps: false,
  }
)

export { Dish }

File: BackendGL/src/models/dish.go
package models

type Dish struct {
	ID          uint   `gorm:"primaryKey" json:"id"`
	Name        string `gorm:"size:255; not null" json:"name"`
	Description string `gorm:"not null" json:"description"`
	MealType    string `gorm:"varchar(20);not null; check:meal_type IN ('cafe_manha', 'almoco', 'cafe_tarde', 'janta')" json:"meal_type"`
	Calories    uint   `json:"calories"`
}

type MenuDish struct {
	ID     uint `gorm:"primaryKey"`
	MenuID uint
	Menu   Menu `gorm:"foreignKey:MenuID;constraint: OnUpdate:CASCADE, OnDelete:CASCADE" json:"menu_dish"`
	DishID uint
	Dish   Dish `gorm:"foreignKey:DishID;constraint: OnUpdate:CASCADE, OnDelete:CASCADE" json:"dish"`
}

type Menu struct {
	ID           uint   `gorm:"primaryKey"`
	Day          string `gorm:"type:date; not null"`
	EnterpriseID uint
	Enterprise   Enterprise `gorm:"foreignKey:EnterpriseID;constraint: OnUpdate:CASCADE, OnDelete:CASCADE" json:"enterprise"`
	Dishes       []Dish     `gorm:"many2many:menu_dish; constraint: OnUpdate:CASCADE, OnDelete:CASCADE" json:"dish"`
}
